tests/absolute_extended_cwd.rs

```diff
@@ -0,0 +1,15 @@
+//! This file tests PathArc::absolute() for Windows when the current directory
+//! uses extended-length path syntax (like `\\?\C:\`).
+extern crate path_abs;
+extern crate tempdir;
+
+// These tests are already run for Unix in absolute_regular_cwd.rs, and Unix
+// doesn't have "extended-length path syntax", so we can make them Windows-only
+// here.
+#[cfg(windows)]
+mod absolute_helpers;
```

Comment:
Just to be clear, the idea here is that `tests/absolute_helpers/mod.rs` contains a bunch of tests that all call `::setup()` at the beginning of each one. Therefore, this integration test includes that module, and defines a top-level `setup()` function that sets up the custom environment for each test to run in.

There's two reasons why this isn't just a bunch of tests that each do their own individual setup:

- We want to run the same tests in both the "regular CWD" and "extended-syntax CWD" situations, and it seemed a shame to copy/paste them, so instead I put them in a shared module.
- cargo defaults to running tests in parallel on multiple threads, but "the current directory" is effectively a global variable. If we ran all the tests in the same process, one test might switch the current directory while another test is running, causing spurious test failures.